tl/features/context_match.py

```diff
@@ -1,32 +1,51 @@
 import pandas as pd
 import re
 import rltk.similarity as similarity
+from tl.exceptions import RequiredInputParameterMissingException
+from statistics import mode
+import gzip
 
 
 class MatchContext(object):
-    def __init__(self, input_path, context_path, args):
+    def __init__(self, input_path, args, context_path=None, custom_context_path=None):
         self.final_data = pd.read_csv(input_path, dtype=object)
         self.data = pd.DataFrame()
         self.result_data = pd.DataFrame()
-        self.context = self.read_context_file(context_path)
+        self.isCustom = False
+        if context_path is None and custom_context_path is None:
+            raise RequiredInputParameterMissingException(
+                'One of the input parameters is required: {} or {}'.format("context_path", "custom_context_path"))
+        elif custom_context_path:
+            self.isCustom = True
```

Comment:
again parameter name can't be camelcased, PEP 8 states that. Please read the standard guidelines